infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -291,7 +301,7 @@ public synchronized void handleMessageLayoutPropose(
 
         log.debug("handleMessageLayoutPropose: New phase 2 rank={},  layout={}",
                 proposeRank, proposeLayout);
-        setPhase2Data(new Phase2Data(proposeRank, proposeLayout));
+        setPhase2Data(new Phase2Data(proposeRank, proposeLayout), msgEpoch);
```

Comment:
Is this correct? The `msgEpoch` can still be different from the epoch in `proposeLayout`, then we will still have the inconsistent epoch issue, right?

Reply:
There is already a comment that describes why this will not happen: https://github.com/CorfuDB/CorfuDB/pull/2075/files#r308951272

Reply:
As discussed, I changed the name to resolve the ambiguity.